src/layout_configuration.cpp

```diff
@@ -0,0 +1,98 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "layout_configuration.hpp"
+
+#include <algorithm>
+#include <sstream>
+#include <vector>
+
+#include "stringutils.hpp"
+
+namespace ovms {
+
+LayoutConfiguration::LayoutConfiguration(const char* layout) :
+    LayoutConfiguration(std::string(layout)) {
```

Comment:
```suggestion
    LayoutConfiguration(layout) {
```
Wouldn't it work?

Reply:
That would end up being endless recursive call:
```
error: constructor delegates to itself
```